core/tests/protractor_utils/general.js

```diff
@@ -256,6 +256,52 @@ var navigateToTopicsAndSkillsDashboardPage = async function() {
   await waitFor.pageToFullyLoad();
 };
 
+var goOffline = async function() {
+  await browser.driver.setNetworkConditions(
+    {
+      offline: true,
+      latency: 0,
+      download_throughput: 0,
+      upload_throughput: 0
+    });
+};
+
+var goOnline = async function() {
+  // Download throughput refers to the maximum number of bytes that can be
+  // downloaded in a given time.
+  // Upload throughput refers to the maximum number of bytes that can be
+  // uploaded in a given time.
+  await browser.driver.setNetworkConditions(
+    {
+      offline: false,
+      latency: 150,
+      download_throughput: 450 * 1024,
+      upload_throughput: 150 * 1024
```

Comment:
Thanks for adding the comment! I'm still unclear on why you chose these particular numbers though. For example, why not an upload throughput of 200 * 1024? Can you explain why you chose these numbers in the comment? I imagine that the exact numbers aren't that important, in which case you could explain what the constraints are.

To add some context, I'm asking for this because we want to avoid "magic numbers" that appear without explanation. Imagine that a year from now, someone reading the code is wondering whether it's okay to change these numbers. We want the comments to answer their question

Reply:
Done.